### PR TITLE
os/fs/smartfs : Apply basic structure & Config of Journaling

### PR DESCRIPTION
**Pull request by Taejun-Kwon**
Add Kconfig & Basic Structure of Journal
Remove unuseful Log

### CODE REVIEW

File: os/fs/driver/mtd/Kconfig [hunk 1]
```diff
@@ -229,6 +229,31 @@ config SMART_CRC_16
...
+
+if MTD_SMART_JOURNALING
+
+config MTD_SMART_JOURNALING_NSECTOR
+	int "Number of reserved logical sector for Journal"
+	default 16
+	---help---
+		Journal sector will be reserved right after smart MTD initialize,
+		It will divided to 2 area to toggle active area.
+		Journal has it's sequence to varify itself.
```
Comment (jsdosa) [inline os/fs/driver/mtd/Kconfig:252]: Is varfiy the right word?

Comment (Taejun-Kwon) [reply os/fs/driver/mtd/Kconfig:252]: yes it means that 'To make  different'

Comment (amogh-samsung) [reply os/fs/driver/mtd/Kconfig:252]: I think varify -> verify?

File: os/fs/smartfs/Kconfig [hunk 2]
```diff
@@ -73,22 +73,14 @@ bool "Dynamic Header"
...
-                Enabling journaling will increase the delay in filesystem
-                operations, because before doing any fs operation, journaling
-                will first log it in journal area. The journal area will also
-                take up space in the flash, leaving lesser space for filesystem.
-                The space reserved for journal area is proportional to max number
-                of file descriptors allowed and sector size defined. Thus, to
-                minimize the area reserved for journaling, it is advised to keep
-                sector size small.
-
+config SMARTFS_SECTOR_RECOVERY
```
Comment (amogh-samsung) [inline os/fs/smartfs/Kconfig:76]: Should the option for recovery be provided under journaling?
Or will we continue to allow it as a separate feature?

File: os/fs/driver/mtd/smart.c [hunk 3]
```diff
@@ -302,8 +301,10 @@ struct smart_struct_s {
 #endif
 #ifdef CONFIG_MTD_SMART_ALLOC_DEBUG
 	size_t bytesalloc;
-	struct smart_alloc_s
-			alloc[SMART_MAX_ALLOCS];	/* Array of memory allocations */
+	struct smart_alloc_s alloc[SMART_MAX_ALLOCS];	/* Array of memory allocations */
+#ifdef CONFIG_MTD_SMART_JOURNALING
```
Comment (amogh-samsung) [inline os/fs/driver/mtd/smart.c:305]: Need to add another #endif statement for this config

File: os/fs/smartfs/Kconfig (hunk 2 above)
Comment (Taejun-Kwon) [reply os/fs/smartfs/Kconfig:76]: Actually Recovery should start from fs layer, so I think let it work separetely for now.

File: os/fs/driver/mtd/Kconfig (hunk 1 above)
Comment (Taejun-Kwon) [reply os/fs/driver/mtd/Kconfig:252]: ok I think this makes confusing, changed comment. please verify grammer and something more.

File: os/fs/driver/mtd/smart.c [hunk 4]
```diff
@@ -378,10 +382,36 @@ struct smart_sect_header_s {
...
+/* State of Journal Log Low 4 bits */
+#define SMART_JOURNAL_STATE_COMMIT   0x01
+#define SMART_JOURNAL_STATE_CHECKED  0x02
+
+/* Type of Journal Data High 4 bits */
+#define SMART_JOURNAL_TYPE_COMMIT    0x10
+#define SMART_JOURNAL_TYPE_RELEASE   0x20
+#define SMART_JOURNAL_TYPE_ERASE     0x40
+
+struct smart_journal_logging_data_s {
```
[the remark this answers is not in the document]
Comment (Taejun-Kwon) [reply os/fs/driver/mtd/smart.c:405]: yeah. If we also save data, entry is more correct I think. (entry + data)
But regarding english, you much better than I.
If you think it is necessary, please raise PR

**Pull request merged**